install/kubernetes/cilium/values.yaml

```diff
@@ -898,7 +898,7 @@ keepDeprecatedProbes: false
 # -- Configure the kube-proxy replacement in Cilium BPF datapath
 # Valid options are "disabled", "probe", "partial", "strict".
 # ref: https://docs.cilium.io/en/stable/gettingstarted/kubeproxy-free/
-kubeProxyReplacement: "probe"
+kubeProxyReplacement: "disabled"
```

Comment:
We usually handle upgrade compatibility inside `install/kubernetes/cilium/templates/cilium-configmap.yaml`. Are you sure this will not override existing user environments when the user runs helm? I suspect this will always apply unless the user has explicitly configured `kubeProxyReplacement`, which may not be the case.

Reply:
I suspect we need a default variable set like this: 

https://github.com/cilium/cilium/blob/519fa82d18e4c59a882c65f078b17dfe7c9a3fbd/install/kubernetes/cilium/templates/cilium-configmap.yaml#L2-L3

Plus wrapping code here that sets either to the default or the user-provided option:

https://github.com/cilium/cilium/blob/519fa82d18e4c59a882c65f078b17dfe7c9a3fbd/install/kubernetes/cilium/templates/cilium-configmap.yaml#L489-L490

Reply:
Actual default value example:

https://github.com/cilium/cilium/blob/519fa82d18e4c59a882c65f078b17dfe7c9a3fbd/install/kubernetes/cilium/templates/cilium-configmap.yaml#L109

Reply:
@joestringer Yes, I've been looking into this. I thought this was still in draft mode because that is unresolved. I'm not sure how to actually guarantee this because we have now set all defaults in the `values.yaml` file.

The upgrade compatibility logic works like this:
```
{{- if hasKey .Values "bpfClockProbe" }}
  enable-bpf-clock-probe: {{ .Values.bpfClockProbe | quote }}
{{- else if eq $defaultBpfClockProbe "true" }}
  enable-bpf-clock-probe: {{ $defaultBpfClockProbe | quote }}
{{- end }}
```

So I thnk for any value that is already defined in `values.yaml`, we can't really provide a new default anymore without also breaking upgrades.

@seanmwinn Thought?

Reply:
One option is to remove the `kube-proxy-replacement` key from the `values.yaml` again.

Reply:
> One option is to remove the kube-proxy-replacement key from the values.yaml again.

This is the approach I've used previously.

Reply:
Done

Reply:
There's no way to override the user intent defined in values but there are ways to use the ordering of values at the command line to override that value by passing it in with --set or with another values file. Helm applies values in the order they are passed in, so the last --set command or values file passed would set the resulting value.

ie. ``helm install -n kube-system cilium cilium/cilium -f values.yaml -f values2.yaml --set kubeProxyReplacement=partial``

Reply:
@seanmwinn the problem is around defaults. If the user doesn't explicitly opt in during the previous release, then if we change the values.yaml value in a new release, they will inherit a configuration change during upgrade. But we don't want that. We want users to stick with the original config defined in the initial install version so that we don't silently break something in their clusters during upgrade.

Reply:
Unless we can tell users to pull all of the exact configuration out of their current cluster during the upgrade process so that we know they are explicitly passing the same options...? We don't have such instructions today.